### PR TITLE
chore: improve 4.0 migrations - part 2

### PR DESCRIPTION
**Pull request by Platonn**
related to #13082 

- [x] TODO: backport to maintenance branch

### CODE REVIEW

File: projects/schematics/src/migrations/4_0/rename-symbol/rename-symbol.ts [hunk 1]
```diff
@@ -412,6 +414,13 @@ export const RENAMED_SYMBOLS_DATA: RenamedSymbol[] = [
     newImportPath: SPARTACUS_USER_ACCOUNT_COMPONENTS,
     // feature-libs/checkout/components/services/express-checkout.service.ts
   },
+  {
+    previousNode: ITEM,
+    previousImportPath: SPARTACUS_STOREFRONTLIB,
+    newNode: ORDER_ENTRY,
+    newImportPath: SPARTACUS_CORE,
+    // projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts
```
Comment (znikola) [inline projects/schematics/src/migrations/4_0/rename-symbol/rename-symbol.ts]: Nitpick: this comment should be before the object ;)

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 2]
```diff
@@ -248,19 +246,19 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
     importPath: SPARTACUS_CORE,
-    comment: `'${SMART_EDIT_MODULE}' was removed. Use @spartacus/smartedit instead.`,
+    comment: `'${SMART_EDIT_MODULE}' was removed. Use @spartacus/smartedit instead. To benefit from lazy loading it by default, consider removing the module import and running the command 'ng add @spartacus/smartedit'.`,
   },
   // projects/core/src/personalization/personalization.module.ts
   {
     node: PERSONALIZATION_MODULE,
     importPath: SPARTACUS_CORE,
-    comment: `'${PERSONALIZATION_MODULE}' was removed. Use @spartacus/tracking/personalization instead.`,
+    comment: `'${PERSONALIZATION_MODULE}' was removed. Use @spartacus/tracking/personalization instead. To benefit from lazy loading it by default, consider removing the module import and running the command 'ng add @spartacus/tracking'.`,
```
Comment (znikola) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: As `Personalization` is just a feature from the tracking lib, maybe we can specify the command like so:
`ng add @spartacus/tracking --features=Personalization`. What do you think?

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 3]
```diff
@@ -248,19 +246,19 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
   },
   // projects/storefrontlib/src/cms-components/product/product-variants/product-variants.module.ts
   {
     node: PRODUCT_VARIANTS_MODULE,
```
Comment (znikola) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts:259]: On line 228 we also have the AsmModule migration. Maybe we should mention the "To benefit from lazy loading it by default, consider removing the module import and running the command..."?

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 4]
```diff
@@ -248,19 +246,19 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
     importPath: SPARTACUS_CORE,
-    comment: `'${PERSONALIZATION_MODULE}' was removed. Use @spartacus/tracking/personalization instead.`,
+    comment: `'${PERSONALIZATION_MODULE}' was removed. Use @spartacus/tracking/personalization instead. To benefit from lazy loading it by default, consider removing the module import and running the command 'ng add @spartacus/tracking --features=Personalization'.`,
   },
   // projects/storefrontlib/src/cms-components/product/product-variants/product-variants.module.ts
   {
     node: PRODUCT_VARIANTS_MODULE,
     importPath: SPARTACUS_STOREFRONTLIB,
-    comment: `'${PRODUCT_VARIANTS_MODULE}' was removed from @spartacus/storefront. Use @spartacus/product/variants feature-library instead.`,
+    comment: `'${PRODUCT_VARIANTS_MODULE}' was removed from @spartacus/storefront. Use @spartacus/product/variants feature-library instead. To benefit from lazy loading it by default, consider removing the module import and running the command 'ng add @spartacus/product'.`,
```
Comment (znikola) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: Similar to the personalization, we can specify the exact feature: `--features=Product-Variants`

**Review by znikola — CHANGES_REQUESTED** [review]

**Review by znikola — APPROVED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**